kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/dates.kt

```diff
@@ -247,3 +248,17 @@ fun Arb.Companion.zonedDateTime(
    localDateTime(minValue, maxValue),
    zoneId
 ) { time, zone -> time.atZone(zone) }
+
+/**
+ * Arberates a stream of random [Date]
+ */
+fun Arb.Companion.date(
+   minDate: String = "1970-01-01",
+   maxDate: String = "2050-12-31",
```

Comment:
working with java.util.Date is inconvenient, is it okay that default params are Strings?

Reply:
What do you think about provide functions to use either Date or String in the API. 

`Arb.date()` feels like a very prioritized API for a class that probably shouldnt be used a lot (`java.util.Date`). What do you think about calling it `Arb.javaDate()`?

Reply:
Yep, renaming to javaDate seems better ✔️ 

Also added second `Arb.javaDate()` function